src/energyplus/Test/AirConditionerVariableRefrigerantFlowFluidTemperatureControlHR_GTest.cpp

```diff
@@ -0,0 +1,449 @@
+  EXPECT_EQ(
+    36,
+    idf_vrf
+      .getDouble(AirConditioner_VariableRefrigerantFlow_FluidTemperatureControl_HRFields::MaximumOutdoorDrybulbTemperatureforDefrostOperation, false)
+      .get());
+  EXPECT_EQ(
+    37, idf_vrf.getDouble(AirConditioner_VariableRefrigerantFlow_FluidTemperatureControl_HRFields::InitialHeatRecoveryCoolingCapacityFraction, false)
+          .get());
+  EXPECT_EQ(
+    38,
+    idf_vrf.getDouble(AirConditioner_VariableRefrigerantFlow_FluidTemperatureControl_HRFields::HeatRecoveryCoolingCapacityTimeConstant, false).get());
+  EXPECT_EQ(
+    39, idf_vrf.getDouble(AirConditioner_VariableRefrigerantFlow_FluidTemperatureControl_HRFields::InitialHeatRecoveryCoolingEnergyFraction, false)
+          .get());
+  EXPECT_EQ(
+    40,
+    idf_vrf.getDouble(AirConditioner_VariableRefrigerantFlow_FluidTemperatureControl_HRFields::HeatRecoveryCoolingEnergyTimeConstant, false).get());
+  EXPECT_EQ(
+    41, idf_vrf.getDouble(AirConditioner_VariableRefrigerantFlow_FluidTemperatureControl_HRFields::InitialHeatRecoveryHeatingCapacityFraction, false)
+          .get());
+  EXPECT_EQ(
+    42,
+    idf_vrf.getDouble(AirConditioner_VariableRefrigerantFlow_FluidTemperatureControl_HRFields::HeatRecoveryHeatingCapacityTimeConstant, false).get());
+  EXPECT_EQ(
+    43, idf_vrf.getDouble(AirConditioner_VariableRefrigerantFlow_FluidTemperatureControl_HRFields::InitialHeatRecoveryHeatingEnergyFraction, false)
+          .get());
+  EXPECT_EQ(
+    44,
+    idf_vrf.getDouble(AirConditioner_VariableRefrigerantFlow_FluidTemperatureControl_HRFields::HeatRecoveryHeatingEnergyTimeConstant, false).get());
+  EXPECT_EQ(45,
+            idf_vrf.getDouble(AirConditioner_VariableRefrigerantFlow_FluidTemperatureControl_HRFields::CompressormaximumdeltaPressure, false).get());
+  EXPECT_EQ(0.5,
+            idf_vrf.getDouble(AirConditioner_VariableRefrigerantFlow_FluidTemperatureControl_HRFields::CompressorInverterEfficiency, false).get());
+  EXPECT_EQ(
+    47,
+    idf_vrf.getDouble(AirConditioner_VariableRefrigerantFlow_FluidTemperatureControl_HRFields::CompressorEvaporativeCapacityCorrectionFactor, false)
+      .get());
+  EXPECT_EQ(
+    3, idf_vrf.getInt(AirConditioner_VariableRefrigerantFlow_FluidTemperatureControl_HRFields::NumberofCompressorLoadingIndexEntries, false).get());
+
+  EXPECT_EQ(3u, idf_vrf.numExtensibleGroups());
+  auto egs = idf_vrf.extensibleGroups();
+  auto loadingIndexes = vrf.loadingIndexes();
+  for (size_t i = 0; i < idf_vrf.extensibleGroups().size(); ++i) {
+    const IdfExtensibleGroup& eg = egs[i];
+    const auto loadingIndex = loadingIndexes[i];
+
+    EXPECT_EQ(i + 1,
+              eg.getDouble(AirConditioner_VariableRefrigerantFlow_FluidTemperatureControl_HRExtensibleFields::CompressorSpeedatLoadingIndex).get());
+    EXPECT_NE("", eg.getString(AirConditioner_VariableRefrigerantFlow_FluidTemperatureControl_HRExtensibleFields::
```

Comment:
1. Is isEmpty when checking for emptyness instead of using a ""
2. here, check the name actually. EXPECT_EQ(evaporativeCapacityMultiplierFunctionofTemperatureCurve1.nameString(), eg.getString(....))

Reply:
👍